### PR TITLE
ES6 and separation of routes

### LINKED ISSUE #9: Split routes into different files
Place under `routes` folder, in a self-named folder with an index.js

e.g.

```
routes
└─ apply
   └─ index.js
```

### LINKED ISSUE #10: Convert codebase to ES2015

### PR DESCRIPTION
**Pull request by troym9731**
closes #9 
closes #10

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+require('babel-core/register');
+var server = require('./server');
```
Comment (therealklanni) [inline app.js]: No need to set this to a var.

File: lib/auth.js [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
-
-var clientId = process.env.CLIENT_ID;
-var clientSecret = process.env.CLIENT_SECRET;
-var nodeEnv = process.env.NODE;
-var baseUrl = nodeEnv === 'production' ?
+import bug from 'debug';
+const debug = bug('SIR:auth');
+import passport from 'passport';
+import githubPassport from 'passport-github2';
+const GitHubStrategy = githubPassport.Strategy;
```
Comment (therealklanni) [inline lib/auth.js]: Lines 4 & 5 can be simplified as `import { Strategy as GitHubStrategy } from 'passport-github2'`

File: lib/helper-functions.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import path from 'path';
```
Comment (therealklanni) [inline lib/helper-functions.js]: Let's just call it `helpers.js` for simplicity.

File: lib/rate-limit.js [hunk 4]
```diff
@@ -1,19 +1,17 @@
...
-    var timestamp = dotty.get(req, 'session.timestamp');
+  return (req, res, next) => {
+    const timestamp = dotty.get(req, 'session.timestamp');
 
-    if (timestamp && Date.now() - timestamp < 3600000) {
+    if (timestamp && Date.now() - timestam < 3600000) {
       res.status(429)
         .send('Whoa, ' + name + ', got a little over-excited there, did ya?');
-      return;
+        return;
```
Comment (therealklanni) [inline lib/rate-limit.js]: Too much indent (got a little over-excited there, did ya?) :stuck_out_tongue_winking_eye:

File: routes/apply/index.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+              x => x,
+              _.partialRight(_.map, (str, i) => {
+                return i ? str : changeCase.title(str);
+              }),
+              _.partial(_.zipObject, ['title', 'value']),
+              _.partialRight(_.assign, { short: true })
+            )
+          )
+            .concat(_.map(files, (file) => (
+              {
```
Comment (therealklanni) [inline routes/apply/index.js]: Let's just collapse the parens and curlies together here:

``` js
({
...
})))
```

File: server.js [hunk 6]
```diff
@@ -1,41 +1,37 @@
...
+import passport from 'passport';
+import session from 'express-session';
+
+import cookieParser from 'cookie-parser';
+import bodyParser from 'body-parser';
+import multer from 'multer';
+import validate from './lib/validate';
+import rateLimit from './lib/rate-limit';
+
+import _ from 'lodash';
```
Comment (therealklanni) [inline server.js]: Please double-check imports. I don't think Lodash is used here, possibly others, too.

File: lib/rate-limit.js [hunk 7]
```diff
@@ -1,12 +1,10 @@
...
+import dotty from 'dotty';
 
 function rateLimit() {
-  return function (req, res, next) {
-    var timestamp = dotty.get(req, 'session.timestamp');
+  return (req, res, next) => {
+    const timestamp = dotty.get(req, 'session.timestamp');
 
-    if (timestamp && Date.now() - timestamp < 3600000) {
+    if (timestamp && Date.now() - timestam < 3600000) {
```
Comment (therealklanni) [inline lib/rate-limit.js]: Oops, what happened? :wink:

Comment (troym9731) [inline lib/rate-limit.js]: Don't know! Haha

File: server.js [hunk 8]
```diff
@@ -53,30 +38,8 @@ if (!env.CLIENT_SECRET) {
...
-
-  strings.apply = _.assign({}, {
-    title: strings.title,
-    gaToken: gaToken
-  }, strings.apply);
-
-  return strings;
-}
+import mod from './lib/slack';
+const slack = mod(slackUrl);
```
Comment (therealklanni) [inline server.js]: Looks like this should have been (re)moved?

**Comment by therealklanni** [comment]
Awesome :metal:

**Pull request merged**